packages/wrangler/src/versions/deploy.ts

```diff
@@ -523,14 +523,19 @@ async function createDeployment(
 
 export function parseVersionSpecs(
 	args: Pick<
-		StrictYargsOptionsToInterface<typeof versionsDeployOptions>,
-		"versionSpecs" | "versionId" | "percentage"
+		VersionsDeployArgs,
+		"_" | "versionSpecs" | "versionId" | "percentage"
 	>
 ): Map<VersionId, OptionalPercentage> {
 	const versionIds: string[] = [];
 	const percentages: OptionalPercentage[] = [];
 
-	for (const spec of args.versionSpecs ?? []) {
+	// args.versionSpecs weren't being parsed in tests, so fallback to args._ if missing
+	const versionSpecs =
+		args.versionSpecs ??
+		args._.filter((arg): arg is string => typeof arg === "string");
+
+	for (const spec of versionSpecs) {
```

Comment:
Had to fallback to `args._` because the way I tried unit-testing this function didn't seem to parse the positional args into `args.versionSpecs` – maybe someone knows a better way?

Reply:
cc @petebacondarwin